pkg/codegen/go/gen_program_utils.go

```diff
@@ -54,3 +54,32 @@ func (p *promptToInputArrayHelper) getPromptItemType() string {
 func (p *promptToInputArrayHelper) getInputItemType() string {
 	return strings.TrimSuffix(p.destType, "Array")
 }
+
+// Provides code for a method which will be placed in the program preamble if deemed
+// necessary. Because many tasks in Go such as reading a file require extensive error
+// handling, it is much prettier to encapsulate that error handling boilerplate as its
+// own function in the preamble.
+func getHelperMethodIfNeeded(functionName string) (string, bool) {
+	var methodBody string
+
+	switch functionName {
+	case "readFile":
+		methodBody =
+			`func readFileOrPanic(path string) pulumi.StringPtrInput {
+				if fileData, err := ioutil.ReadFile(path); err == nil {
+					return pulumi.String(string(fileData[:]))
+				} else {
+					panic(err.Error())
+				}
+			}`
+	case "sha1":
+		methodBody =
+			`func sha1(input string) pulumi.StringPtrInput {
+				inputBytes := []byte(input)
+				return pulumi.String(fmt.Sprintf("%x", sha1.Sum(inputBytes)))
+			}`
```

Comment:
Interestingly, this one requires no error handling, so could have just been inlined. But most likely nicer to read the way you have structured this.

Reply:
I thought the same. If the inlined method ends up being too long, moving it into a helper is preferable for the users of the example.